Sources/SwiftFormatPrettyPrint/PrettyPrint.swift

```diff
@@ -494,7 +517,13 @@ public class PrettyPrinter {
       // of any prior break tokens.
       case .newlines:
         if let index = delimIndexStack.last, case .break = tokens[index] {
-          lengths[index] += total
+          if index == i - 1 {
+            // A break immediately preceding a newline should have a length of zero, so that it
+            // doesn't fire.
+            lengths[index] = 0
```

Comment:
Is this to avoid dealing with a newline after a break fires? I'm wondering if that could still be an issue in cases where the break is encountered after some printed tokens that **must** exceed the maximum line length (e.g. a really long function name in a function decl).

Reply:
That seems to work ok; the following example breaks correctly, with the curly brace even being reset appropriately. Is this the kind of situation you were thinking of?

```swift
func test_test_test_test_test_test_test_test_test_test_test_test_test_test_test_test_test_test_test_test_test_test_test_test(foo: Int) {
  foo()
  bar()
}

// becomes

func
  test_test_test_test_test_test_test_test_test_test_test_test_test_test_test_test_test_test_test_test_test_test_test_test(
    foo: Int
  )
{
  foo()
  bar()
}
```

Reply:
Yup, that looks good. Thanks for checking!